core/domain/suggestion_services.py

```diff
@@ -542,3 +555,43 @@ def check_can_resubmit_suggestion(suggestion_id, user_id):
     suggestion = get_suggestion_by_id(suggestion_id)
 
     return suggestion.author_id == user_id
+
+
+def check_can_edit_suggestion(user, suggestion_id):
+    """Checks whether the user have permission to edit suggestion or not.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        suggestion_id: str. Id of a suggestion.
+
+    Returns:
+        users_which_have_rights: dict(str, str). Dict whose key value pairs
```

Comment:
Make sure to proofread your changes. This function returns only bools from what I see.

Reply:
ohh, sorry for that, i missed it.

Reply:
done